src/commands/create.ts

```diff
@@ -71,59 +98,143 @@ export default class Create extends Command {
-      )
-    );
+    if (!flags.profileId && !flags.providerName && !flags.interactive) {
+      throw userError('Invalid command! Specify profileId or providerName', 1);
+    }
+    if (flags.path && !(await exists(flags.path))) {
+      throw userError(
+        `Invalid command! Path "${flags.path}" does not exist`,
+        1
+      );
+    }
+
+    //Not creating anything
+    if (!flags.provider && !flags.profile && !flags.map && !flags.interactive) {
+      this.warn(
+        'Create command without flag --profile or --provider or --map or --interactive does not do anything'
+      );
+
+      return;
+    }
+
+    //Interactive
+    if (flags.interactive) {
+      flags.profile = await this.confirmPrompt(
+        'Do you want to create a profile?'
+      );
+      flags.map = await this.confirmPrompt('Do you want to create a map?');
+      flags.provider = await this.confirmPrompt(
+        'Do you want to create a provider?'
+      );
+      //We need profile Id
+      if (!flags.profileId && (flags.profile || flags.map)) {
+        let profileInput: string | undefined = undefined;
+        profileInput = await this.inputPrompt(
+          'Enter profile Id in format [scope](optional)/[name]'
+        );
+        if (!profileInput) {
+          throw userError('Invalid command! Profile Id must be defined', 1);
+        }
+        flags.profileId = profileInput;
+      }
+      //We need provider name
+      if (!flags.providerName && (flags.provider || flags.map)) {
+        flags.providerName = [];
+        let priority = 1;
+        let exit = false;
+        let providerInput: string | undefined = undefined;
+        const priorityToString: Map<number, string> = new Map([
+          [1, 'primary'],
+          [2, 'secondary'],
+          [3, 'third'],
+          [4, 'fourth'],
+          [5, 'fifth'],
+        ]);
+        while (!exit) {
+          providerInput = await this.inputPrompt(
+            `Enter provider name of ${
+              priorityToString.get(priority) || priority
+            } provider.\nExit loop by pressing enter without any input.`
+          );
+          if (!providerInput) {
+            //We don't have any name
+            if (priority === 1) {
+              throw userError(
+                'Invalid command! At least one provider must be defined',
+                1
+              );
+            }
+            exit = true;
+            continue;
+          }
+          flags.providerName.push(providerInput);
+          priority++;
+        }
+      }
+    }
 
     if (flags.quiet) {
       this.logCallback = undefined;
       this.warnCallback = undefined;
     }
+    let profileId: string | undefined = undefined;
+    let providerNames: string[] = [];
 
-    if (argv.length > 2) {
-      throw userError('Invalid command!', 1);
+    //Check inputs
```

Comment:
These checks don't have to be here if you add `dependsOn` array in definition of `--profile`, `--map` and `--provider` flags.

Reply:
Sadly the do. Only way (I can think of) to make this work is to add `excludes` between `interactive` and `--profile`, `--map`, `--provider`, `providerName` and `--profileId`